assets/js/main.js

```diff
@@ -534,11 +539,91 @@ document.addEventListener('DOMContentLoaded', function(){
         });
     });
 
+    // Related Posts
+    function displayRelatedPosts(pages){
+        const refBox = document.getElementById('related-box');
+
+        if (!refBox) return;
+
+        var relatedPosts = [];
+        var currPost = pages.find(obj => {return obj.url === location.pathname});
+
+        let currTags = currPost.tags.split(', ');
+        let currCategory = currPost.path.split(' > ').pop();
+
+        for (var i = 0; i < pages.length; i++) {
+            let page = pages[i];
+
+            if (page.type === 'category') continue;
+
+            if (page.title === currPost.title) continue;
+
+            let tags = page.tags.split(', ');
+            let category = page.path.split(' > ').pop();
+            let correlationScore = 0;
+
+            for (var j = 0; j < currTags.length; j++){
+                if (tags.indexOf(currTags[j]) != -1) correlationScore += 1;
+            }
+
+            if (category === currCategory) correlationScore += 1;
+
+            if (correlationScore == 0) continue;
+
+            relatedPosts.push({
+                'title': page.title,
+                'date': page.date,
+                'category': category,
+                'url': page.url,
+                'thumbnail': page.image,
+                'score': correlationScore
+            });
+        }
+
+        relatedPosts.sort(function (a, b) {
+            if(a.hasOwnProperty('score')){
+                return b.score - a.score;
+            }
+        });
+
+        if (relatedPosts.length == 0){
+            $('#related-box').hide();
+            return;
+        }
+
+        for (var i = 0; i < Math.max(relatedPosts.length, 6); i++){
+            let post = relatedPosts[i];
+            let date = new Date(post.date);
+            date = date.toLocaleString('en-US', {day: 'numeric', month:'long', year:'numeric'});
+
+            if (post.thumbnail === ''){
+                post.thumbnail = "/assets/img/thumbnail/empty.jpg";
+            }
+
+            $('#related-posts').append(
+                '<li class="related-item"><a href="' + post.url +
+                    '"><img src="' + post.thumbnail + 
+                    '"/><p class="category">' + post.category +  
```

Comment:
Please assure that rendering is not failed even when the `post.category` is blank.

Reply:
I suspect, it could arise an error earlier in `line 562`